staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/validation.go

```diff
@@ -70,6 +70,8 @@ func validate(ctx context.Context, pth *field.Path, s *structuralschema.Structur
 		rootSchema = s
 	}
 
+	isRoot := s == rootSchema
```

Comment:
what does isRoot control? is setting this to `true` correct if this is validating an `XEmbeddedResource` nested down in the depths of a CR (because `rootSchema = s` on line 70 will make this evaluate to true)

Reply:
put another way, does hard-coding this to `true` or to `false` fail any tests? if not, it probably should

Reply:
If true the CEL "decl" (CELs type information) is guaranteed by `celmodel.WithTypeAndObjectMeta()` to contain kind, apiVersion, metadata.name and metadata.generateName properties.  I believe we want this to be true both for the root object of a CRD and for all `XEmbeddedResource` (the godoc for SchemaDeclType is also written to indicate that we should set it in both cases).

Hard coding either true or false does fail tests (I ended up fixing this because I had introduced a code defect where it became true everywhere and then started tracing through logic.  Setting it false everywhere should also fail some tests that access the metadata fields, I'll verify this).